src/Microsoft.DotNet.Watcher.Core/Internal/ProjectWatcher.cs

```diff
@@ -107,6 +107,10 @@ private void GetProjectFilesClosure(string projectFile, ISet<string> closure)
                 foreach (var file in project.Files)
                 {
                     closure.Add(file);
+
+                    // We need to add the folder because folder deletions only trigger 
+                    // for the folder, not for the files inside it
+                    closure.Add(Path.GetDirectoryName(file));
```

Comment:
Have you checked if watching the folder causes any triggers to be fired for any changes to the files underneath it, including the excluded files?